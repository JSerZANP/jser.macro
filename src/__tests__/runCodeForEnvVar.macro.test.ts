import { pluginTester } from "babel-plugin-tester";
import plugin from "babel-plugin-macros";

pluginTester({
  plugin,
  snapshot: true,
  filepath: __dirname,
  formatResult: (r) => r,
  tests: [
    {
      title: "throws if no arguments",
      snapshot: false,
      error: true,
      code: `
        import {runCodeForEnvVar} from '../dist/runCodeForEnvVar.macro'
        runCodeForEnvVar()
      `,
    },
    {
      title: "throws if only 1 argument",
      snapshot: false,
      error: true,
      code: `
        import {runCodeForEnvVar} from '../dist/runCodeForEnvVar.macro'
        runCodeForEnvVar('ENABLE_DEBUG')
      `,
    },
    {
      title: "throws if only first arg is not string literal",
      snapshot: false,
      error: true,
      code: `
        import {runCodeForEnvVar} from '../dist/runCodeForEnvVar.macro'
        runCodeForEnvVar(true, () => {})
      `,
    },
    {
      title: "throws if only first arg is variable",
      snapshot: false,
      error: true,
      code: `
        import {runCodeForEnvVar} from '../dist/runCodeForEnvVar.macro'
        const variable = 'true';
        runCodeForEnvVar(variable, () => {})
      `,
    },
    {
      title: "throws if 2nd arg us not function expression",
      snapshot: false,
      error: true,
      code: `
        import {runCodeForEnvVar} from '../dist/runCodeForEnvVar.macro'
        runCodeForEnvVar('ENABLE_DEBUG', true)
      `,
    },
    {
      title: "keep the code in IIFE if target env var is 'true'",
      setup() {
        process.env.ENABLE_DEBUG = "true";
      },
      code: `
        import {runCodeForEnvVar} from '../dist/runCodeForEnvVar.macro'
        runCodeForEnvVar('ENABLE_DEBUG', () => {
          console.log('ENABLE_DEBUG enabled')
        })
      `,
    },
    {
      title: "strip the code if the target env var doesn't exist",
      setup() {
        process.env.ENABLE_DEBUG = "";
      },
      code: `
        import {runCodeForEnvVar} from '../dist/runCodeForEnvVar.macro'
        runCodeForEnvVar('ENABLE_DEBUG', () => {
          console.log('ENABLE_DEBUG enabled')
        })
      `,
    },
    {
      title: "strip the code if the target env var doesn't match",
      setup() {
        process.env.ENABLE_DEBUG = "SOME_VALUE";
      },
      code: `
        import {runCodeForEnvVar} from '../dist/runCodeForEnvVar.macro'
        runCodeForEnvVar('ENABLE_DEBUG', () => {
          console.log('ENABLE_DEBUG enabled')
        })
      `,
    },
    {
      title: "support multiple calls",
      setup() {
        process.env.ENABLE_DEBUG = "true";
      },
      code: `
        import {runCodeForEnvVar} from '../dist/runCodeForEnvVar.macro'
        runCodeForEnvVar('ENABLE_DEBUG', () => {
          console.log('ENABLE_DEBUG enabled! 1')
        })
        runCodeForEnvVar('ENABLE_DEBUG', () => {
          console.log('ENABLE_DEBUG enabled! 2')
        })
      `,
    },
    {
      title: "support multiple calls with multiple env vars",
      setup() {
        process.env.ENABLE_DEBUG = "true";
        process.env.ENABLE_ADVANCED_LOGGING = "true";
        process.env.ENABLE_AB_TEST = "false";
      },
      code: `
        import {runCodeForEnvVar} from '../dist/runCodeForEnvVar.macro'
        runCodeForEnvVar('ENABLE_DEBUG', () => {
          console.log('ENABLE_DEBUG enabled! 1')
        })
        runCodeForEnvVar('ENABLE_ADVANCED_LOGGING', () => {
          console.log('ENABLE_ADVANCED_LOGGING enabled! 2')
        })
        runCodeForEnvVar('ENABLE_AB_TEST', () => {
          console.log('ENABLE_AB_TEST enabled')
        })
      `,
    },
  ],
});
