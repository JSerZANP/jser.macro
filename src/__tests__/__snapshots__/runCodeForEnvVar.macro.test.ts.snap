// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`macros > 6. keep the code in IIFE if target env var is 'true' > 6. keep the code in IIFE if target env var is 'true' 1`] = `


import runCodeForEnvVar from '../dist/runCodeForEnvVar.macro'
runCodeForEnvVar('ENABLE_DEBUG', () => {
  console.log('ENABLE_DEBUG enabled')
})
      

      ↓ ↓ ↓ ↓ ↓ ↓

(() => {
  console.log('ENABLE_DEBUG enabled');
})();

`;

exports[`macros > 7. strip the code if the target env var doesn't exist > 7. strip the code if the target env var doesn't exist 1`] = `


import runCodeForEnvVar from '../dist/runCodeForEnvVar.macro'
runCodeForEnvVar('ENABLE_DEBUG', () => {
  console.log('ENABLE_DEBUG enabled')
})
      

      ↓ ↓ ↓ ↓ ↓ ↓



`;

exports[`macros > 8. strip the code if the target env var doesn't match > 8. strip the code if the target env var doesn't match 1`] = `


import runCodeForEnvVar from '../dist/runCodeForEnvVar.macro'
runCodeForEnvVar('ENABLE_DEBUG', () => {
  console.log('ENABLE_DEBUG enabled')
})
      

      ↓ ↓ ↓ ↓ ↓ ↓



`;

exports[`macros > 9. support multiple calls > 9. support multiple calls 1`] = `


import runCodeForEnvVar from '../dist/runCodeForEnvVar.macro'
runCodeForEnvVar('ENABLE_DEBUG', () => {
  console.log('ENABLE_DEBUG enabled! 1')
})
runCodeForEnvVar('ENABLE_DEBUG', () => {
  console.log('ENABLE_DEBUG enabled! 2')
})
      

      ↓ ↓ ↓ ↓ ↓ ↓

(() => {
  console.log('ENABLE_DEBUG enabled! 1');
})();
(() => {
  console.log('ENABLE_DEBUG enabled! 2');
})();

`;

exports[`macros > 10. support multiple calls with multiple env vars > 10. support multiple calls with multiple env vars 1`] = `


import runCodeForEnvVar from '../dist/runCodeForEnvVar.macro'
runCodeForEnvVar('ENABLE_DEBUG', () => {
  console.log('ENABLE_DEBUG enabled! 1')
})
runCodeForEnvVar('ENABLE_ADVANCED_LOGGING', () => {
  console.log('ENABLE_ADVANCED_LOGGING enabled! 2')
})
runCodeForEnvVar('ENABLE_AB_TEST', () => {
  console.log('ENABLE_AB_TEST enabled')
})
      

      ↓ ↓ ↓ ↓ ↓ ↓

(() => {
  console.log('ENABLE_DEBUG enabled! 1');
})();
(() => {
  console.log('ENABLE_ADVANCED_LOGGING enabled! 2');
})();

`;
